jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -150,21 +150,21 @@ object CodeWhispererUtil {
         return connection.startUrl
     }
 
-    private fun tokenProvider(project: Project) = (
+    private fun tokenProvider() = (
         ToolkitConnectionManager
-            .getInstance(project)
+            .getInstance(null)
             .activeConnectionForFeature(CodeWhispererConnection.getInstance()) as? AwsBearerTokenConnection
         )
         ?.getConnectionSettings()
         ?.tokenProvider
         ?.delegate as? BearerTokenProvider
 
-    fun reconnectCodeWhisperer(project: Project) {
-        val connection = ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(CodeWhispererConnection.getInstance())
+    fun reconnectCodeWhisperer() {
+        val connection = ToolkitConnectionManager.getInstance(null).activeConnectionForFeature(CodeWhispererConnection.getInstance())
         if (connection !is BearerSsoConnection) return
         ApplicationManager.getApplication().executeOnPooledThread {
             getConnectionStartUrl(connection)?.let { startUrl ->
-                loginSso(project, startUrl, requestedScopes = connection.scopes)
+                loginSso(null, startUrl, requestedScopes = connection.scopes)
```

Comment:
There's also one loginSso in the ToolkitConnectionDialog -- does that need to be creating an application-wide cwspr connection as well?
Meanwhile we only want to do this for the CodeWhisperer path( The `Start` under CodeWhisperer node), not when users click open the dialog by clicking on the `...` and click `Add Connection`

Reply:
updated